assets/javascripts/discourse/initializers/discourse-calendar.js

```diff
@@ -629,19 +741,23 @@ function initializeDiscourseCalendar(api) {
     return defaultTimezone || currentUser?.timezone || moment.tz.guess();
   }
 
-  function _setupTimezonePicker(calendar, timezone) {
+  function _setupTimezonePicker(calendar, timezone, resetDynamicEvents) {
     const tzPicker = document.querySelector(
       ".discourse-calendar-timezone-picker"
     );
     if (tzPicker) {
       tzPicker.addEventListener("change", function (event) {
         calendar.setOption("timeZone", event.target.value);
+        resetDynamicEvents();
         _insertAddToCalendarLinks(calendar);
       });
 
-      moment.tz.names().forEach((tz) => {
-        tzPicker.appendChild(new Option(tz, tz));
-      });
+      moment.tz
+        .names()
+        .filter((t) => !t.startsWith("Etc/GMT"))
```

Comment:
Curious, why do we need this?

Reply:
I added that comment to the commit

These options (Etc/GMT-04, Etc/GMT+5, etc.) are being parsed incorrectly.
They are essentially being parsed as if the signs were flipped.
I tried forcefully feeding the timezone with the opposite sign,
which fixed the event rendering but not the calendar itself.
I'm opting to remove them from the list to avoid confusion.